lesson7/main.js

```diff
@@ -1,34 +1,47 @@
 "use strict";
 
+const wrapper = document.getElementById("js-wrapper");
 const ul = document.getElementById("js-lists");
 const fragment = document.createDocumentFragment();
-const fetchedData = new Promise((resolve, reject) => {
-  const listItems = [
-    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
-    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
-  ];
-  setTimeout(() => {
-    resolve(listItems);
-  }, 3000);
-});
 
-function createNewList(data) {
-  data.forEach((key) => {
-    const li = document.createElement("li");
-    const a = document.createElement("a");
-    const img = document.createElement("img");
+window.onload = () => {
+  const loader = document.createElement("div");
+  const loaderImage = document.createElement("img");
+  loader.classList.add("loading");
+  loaderImage.src = "img/loading-circle.gif";
+  wrapper.appendChild(loader).appendChild(loaderImage);
 
-    a.href = `/${key.to}`;
-    a.textContent = key.text;
-    img.src = key.img;
-    img.alt = key.alt;
-
-    fragment
-      .appendChild(li)
-      .appendChild(a)
-      .insertAdjacentElement("afterbegin", img);
+  const fetchedData = new Promise((resolve, reject) => {
+    const listItems = [
+      { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+      { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+    ];
+    setTimeout(() => {
+      resolve(listItems);
+    }, 2500);
   });
-  ul.appendChild(fragment);
-}
 
-fetchedData.then((listItems) => createNewList(listItems));
+  function createNewList(data) {
+    data.forEach((key) => {
+      const li = document.createElement("li");
+      const a = document.createElement("a");
+      const img = document.createElement("img");
+
+      a.href = `/${key.to}`;
+      a.textContent = key.text;
+      img.src = key.img;
+      img.alt = key.alt;
+
+      fragment
+        .appendChild(li)
+        .appendChild(a)
+        .insertAdjacentElement("afterbegin", img);
+    });
+    ul.appendChild(fragment);
+  }
+
+  fetchedData.then((listItems) => {
+    loader.classList.add("loaded");
```

Comment:
You can cut it out as a function as well.

Reply:
I implemented some functions with your advice for:
1 on loading 
2 finish loading

If I misunderstood what you meant, please kindly tell me!
[d0c353e](https://github.com/yuka830/js-lessons/pull/9/commits/d0c353ed3e5999fc2537fe567333f29507353c70)

Reply:
Thanks for the fix. The way you defined the function, perfect!!!👍